pkgs/by-name/hi/himalaya/package.nix

```diff
@@ -3,65 +3,71 @@
-  inherit buildNoDefaultFeatures buildFeatures;
+let
+  version = "1.0.0";
+  hash = "sha256-NpkjnU4WBHnXHBoFqkMmX1l2xb038dDYsvWjNnpzBnM=";
+  cargoHash = "sha256-pIrNSCqlRsx2QlknSVSGvkfoHenpzfTd1AAVywBzdpo=";
+in
+
+rustPlatform.buildRustPackage {
+  inherit version cargoHash;
 
   pname = "himalaya";
-  version = "1.0.0-beta.4";
 
   src = fetchFromGitHub {
-    owner = "soywod";
-    repo = pname;
+    inherit hash;
+    owner = "pimalaya";
+    repo = "himalaya";
     rev = "v${version}";
-    hash = "sha256-NrWBg0sjaz/uLsNs8/T4MkUgHOUvAWRix1O5usKsw6o=";
   };
 
-  cargoHash = "sha256-YS8IamapvmdrOPptQh2Ef9Yold0IK1XIeGs0kDIQ5b8=";
-
-  NIX_LDFLAGS = lib.optionals stdenv.hostPlatform.isDarwin [
-    "-F${darwin.apple_sdk.frameworks.AppKit}/Library/Frameworks"
-    "-framework"
-    "AppKit"
-  ];
+  buildNoDefaultFeatures = withDefaultFeatures;
+  buildFeatures = withFeatures;
 
   nativeBuildInputs = [ pkg-config ]
-    ++ lib.optional (builtins.elem "pgp-gpg" buildFeatures) pkg-config
     ++ lib.optional (installManPages || installShellCompletions) installShellFiles;
 
   buildInputs = [ ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin (with darwin.apple_sdk.frameworks; [ AppKit Cocoa Security ])
-    ++ lib.optional (builtins.elem "notmuch" buildFeatures) notmuch
-    ++ lib.optional (builtins.elem "pgp-gpg" buildFeatures) gpgme;
+    ++ lib.optional stdenv.hostPlatform.isDarwin apple-sdk
+    ++ lib.optional (builtins.elem "notmuch" withFeatures) notmuch
+    ++ lib.optional (builtins.elem "pgp-gpg" withFeatures) gpgme;
+
+  # most of the tests are lib side
+  doCheck = false;
 
-  postInstall = lib.optionalString installManPages ''
-    mkdir -p $out/man
-    $out/bin/himalaya man $out/man
-    installManPage $out/man/*
+  postInstall = ''
+    mkdir -p $out/share/{applications,completions,man}
+    cp assets/himalaya.desktop "$out"/share/applications/
+  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    "$out"/bin/himalaya man "$out"/share/man
+  '' + lib.optionalString installManPages ''
+    installManPage "$out"/share/man/*
+  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    "$out"/bin/himalaya completion bash > "$out"/share/completions/himalaya.bash
+    "$out"/bin/himalaya completion elvish > "$out"/share/completions/himalaya.elvish
+    "$out"/bin/himalaya completion fish > "$out"/share/completions/himalaya.fish
+    "$out"/bin/himalaya completion powershell > "$out"/share/completions/himalaya.powershell
+    "$out"/bin/himalaya completion zsh > "$out"/share/completions/himalaya.zsh
   '' + lib.optionalString installShellCompletions ''
-    installShellCompletion --cmd himalaya \
-      --bash <($out/bin/himalaya completion bash) \
-      --fish <($out/bin/himalaya completion fish) \
-      --zsh <($out/bin/himalaya completion zsh)
+    installShellCompletion "$out"/share/completions/himalaya.{bash,fish,zsh}
   '';
 
-  meta = with lib; {
+  meta = rec {
```

Comment:
Nit, but I don't think most would consider the use of `rec` in meta good practice